crates/ide_completion/src/render.rs

```diff
@@ -608,6 +631,10 @@ impl S {
                         documentation: Documentation(
                             "Method docs",
                         ),
+                        relevance: CompletionRelevance {
+                            exact_name_match: false,
+                            exact_type_match: true,
```

Comment:
Perhaps we want a slight relevance penalty for a recursive call? Or else if you are typing at the end of a function you will always be suggested the recursive call with high relevance, due to `exact_type_match`.

Reply:
I think it's fine to roll without that for a time being, as the exact rule here seems to be fiddly.